app/controllers/api/admin/articles_controller.rb

```diff
@@ -9,6 +9,21 @@ def index
       render json: article, each_serializer: Admin::Article::IndexSerializer
   end
 
+  def update
+    if params[:activity] == "PUBLISH"
+      begin
+        article = Article.find(params[:id])
+        article.premium = params[:premium] || article.premium
+        article.category = params[:category] || article.category
+        article.published = true
+        article.save
+        render json: { message: 'Article successfully published!'}
+      rescue => e
+        render json: { message: "Article not published: " + e.to_s }, status: 422
+      end
+    end
+  end
+
```

Comment:
What does  {+e.to_s}  mean?

Reply:
It is the error, and it converts it to_string, so it gets the message from the error.

Reply:
And the plus appends it to the previous string. You can see the results in the test file.

Reply:
Personally I think string interpolation/templating makes more sense when reading it.

Reply:
You could also do e.message, which is a bit more readable.

Reply:
https://apidock.com/rails/ActiveModel/Errors/full_messages

Reply:
Addressed this and made it more clear.